src/App.tsx

```diff
@@ -1,24 +1,75 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Loader } from './components/Loader';
+import { AppContext } from './components/AppContext';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
+
+  // eslint-disable-next-line no-unused-vars
+  const {
+    userId,
+    allTodos,
+    setAllTodos,
+    setShouldShowError,
+    showError,
```

Comment:
I believe setShouldShowError should be named as setShowError

Reply:
`shouldShowError` has boolean type and decides if error is shown
`setShouldShowError` is used to change the value of the state

`showError` is just a helper function that changes `shouldShowError` state value and sets a timer

![image](https://user-images.githubusercontent.com/78492113/233151656-3cc4ce5a-4393-49e9-9596-38cdec1e2209.png)
![image](https://user-images.githubusercontent.com/78492113/233151722-23e2670f-4a19-4c56-b0d7-9a4561f1ff45.png)